exporter/collector/metrics.go

```diff
@@ -952,6 +964,31 @@ func numberDataPointToValue(
 		metricpb.MetricDescriptor_DOUBLE
 }
 
+// This function maps the values of the metric to certain types which are supported in Google Cloud Monitoring, but not in OTEL.
+// Such types includes STRING and BOOL. The conversion only happens for metric kind GAUGE and only under presence of certain attributes in the metric attributes.
+// The function returns the converted value and type if conditions are met, otherwise a nil value with value type MetricDescriptor_VALUE_TYPE_UNSPECIFIED is returned.
```

Comment:
Consider returning an error here instead to represent the exception case. It may be cleaner not sure